### ISSUE TITLE
FlammableBlockRegistry ignores tags unless `/reload`

### ISSUE THREAD

**Issue by SolidBlock-cn** [FabricMC/fabric#2108]
When adding a block tag to `FlammableBlockRegistry`, it is ignored, which means blocks in this tag cannot be flammable. If you `/reload`, it can be normal, even if you re-enter the world. But when you re-launch the game, it will happen again.

```
minecraft_version=1.18.1
yarn_mappings=1.18.1+build.22
loader_version=0.13.3
fabric_version=0.46.6+1.18
```

**Comment by Technici4n** [comment]
Can't reproduce in 1.19. Please ensure that you are registering the tag early enough (i.e. in your mod initializer), and let me know if the issue persists.

**Comment by Technici4n** [comment]
Looking at the code, it seems that it should be an issue though... I am a bit confused. 😅

**Comment by Technici4n** [comment]
This is indeed an issue, I think I was unable to reproduce this on Fabric because of the experimental settings warning causing the resources to be loaded a second time... The simplest fix would require #2265 to be merged, so I could use the TagsLoaded event.
